consent/api/v1/views.py

```diff
@@ -192,14 +199,23 @@ def post(self, request):
             return Response({'error': str(invalid_request)}, status=HTTP_400_BAD_REQUEST)
 
         user_id = get_user_id(username)
-        self.set_consent_state(True, user_id, course_id, enterprise_customer_uuid)
-        provided = consent_provided(user_id, course_id, enterprise_customer_uuid)
         required = consent_required(user_id, course_id, enterprise_customer_uuid)
+        exists = consent_exists(user_id, course_id, enterprise_customer_uuid)
+
+        if not enterprise_customer.catalog_contains_course_run(request.user, course_id):
+            required = False
+
+        if required:
+            self.set_consent_state(True, user_id, course_id, enterprise_customer_uuid)
+            provided = consent_provided(user_id, course_id, enterprise_customer_uuid)
+            required = consent_required(user_id, course_id, enterprise_customer_uuid)
```

Comment:
@haikuginger OK, let's add a comment that mentions this. Otherwise this will continue to look like an opportunity for refactoring to anyone not familiar with the logic in `set_consent_state`.